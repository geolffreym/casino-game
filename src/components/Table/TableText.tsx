import styled from 'styled-components'

const TableText = styled.span<{ color: string }>`
  font-weight: bold;
  color: ${({ color }) => color};
`

export default TableText
