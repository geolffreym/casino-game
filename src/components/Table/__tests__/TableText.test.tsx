import TableText from '@components/Table/TableText'
import { shallow } from 'enzyme'
import 'jest-styled-components'
import React from 'react'

describe('<TableText />', () => {
  it('should render table text', () => {
    const component = shallow(<TableText color='#999999'>$ 10 USD</TableText>)
    expect(component).toMatchSnapshot()
  })

  it('should have table amount children text', () => {
    const component = shallow(<TableText color='#999999'>$ 10 USD</TableText>)

    expect(component.text()).toEqual('$ 10 USD')
  })

  it('should have table text valid color styles', () => {
    const component = shallow(<TableText color='#999999' />)
    expect(component).toHaveStyleRule('color', '#999999')
  })
})
