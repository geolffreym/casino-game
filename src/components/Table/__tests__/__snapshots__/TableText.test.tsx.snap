// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TableText /> should render table text 1`] = `ShallowWrapper {}`;
